fs/nfs/nfs_vfsops.c

```diff
@@ -2035,13 +2014,13 @@ static int nfs_bind(FAR struct inode *blkdriver, FAR const void *data,
 
   /* Initialize the allocated mountpt state structure. */
 
-  /* Initialize the semaphore that controls access.  The initial count
-   * is zero, but nfs_semgive() is called at the completion of
+  /* Initialize the mutex that controls access.
+   * nxmutex_lock() is called at the completion of
    * initialization, incrementing the count to one.
    */
 
-  nxsem_init(&nmp->nm_sem, 0, 0);   /* Initialize the semaphore that
-                                     * controls access */
+  nxmutex_init(&nmp->nm_lock);   /* Initialize the mutex that
```

Comment:
ditto. do we need to `nxmutex_destroy(&nmp->nm_lock);` if `rpc = (FAR struct rpcclnt *)kmm_zalloc(sizeof(struct rpcclnt));` fails